howso/engine/trainee.py

```diff
@@ -70,35 +88,33 @@ class Trainee(BaseTrainee):
     ----------
     name : str, optional
         The name of the trainee.
-    features : dict of {str: dict}
-        The feature attributes of the trainee. Where feature `name` is the key
-        and a sub dictionary of feature attributes is the value.
+    features : SingleTableFeatureAttributes, optional
+        The feature attributes of the trainee. Where feature ``name`` is the key
+        and a sub dictionary of feature attributes is the value. If this is not
+        specified in the constructor, it must be set during or before :meth:`train`.
     default_action_features : list of str, optional
         The default action feature names of the trainee.
     default_context_features : list of str, optional
         The default context feature names of the trainee.
     id : str, optional
         The unique identifier of the Trainee. The client automatically completes
         this field and the user should NOT manually use this parameter. Please use
-        the `name` parameter to manually specify a Trainee name.
-    library_type : str, optional
-        The library type of the Trainee. Valid options include:
-
-            - "st": use single-threaded library.
-            - "mt": use multi-threaded library.
-    max_wait_time : int or float, default 30
+        the ``name`` parameter to manually specify a Trainee name.
+    library_type : {"st", "mt"}, optional
+        The library type of the Trainee. "st" will use the single-threaded library,
+        while "mt" will use the multi-threaded library.
+    max_wait_time : int | float, default 30
         The number of seconds to wait for a trainee to be created and become
-        available before aborting gracefully. Set to `0` (or None) to wait as
+        available before aborting gracefully. Set to ``0`` (or None) to wait as
         long as the system-configured maximum for sufficient resources to
         become available, which is typically 20 minutes.
-    persistence : str, default "allow"
-        The requested persistence state of the trainee. Allowed values include
-        "allow", "always", and "never".
-    project : str or Project, optional
+    persistence : {"allow", "always", "never"}, default "allow"
```

Comment:
I don't hate this, but wouldn't `{"A" | "B" | "C"}` be a better way to express "one of these options"? Just a discussion point, I don't think using commas here expresses mutual-exclusivity well.